### PR TITLE
Create Menu class, Type class and mapping into SQL server

### PR DESCRIPTION
**Pull request by PNH00**

### CODE REVIEW

**Comment by PNH00** [comment]
[Create Menu class, Type class and mapping into SQL server](https://trello.com/c/1vIHh5ia/14-create-menu-class-type-class-and-mapping-into-sql-server)

File: RestaurentManagement/src/main/java/com/restapi/models/Menu.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    private UUID id;
+    @Column(nullable = false,columnDefinition = "nvarchar(50)")
+    private String name;
+    @Column(nullable = false,columnDefinition = "nvarchar(50)")
+    private String description;
+    @Column(nullable = false,columnDefinition = "nvarchar(150)")
+    private String image;
+    @Column(nullable = false)
+    private double price;
+
```
Comment (vtronghieu) [inline RestaurentManagement/src/main/java/com/restapi/models/Menu.java]: Pls remove this blank

File: RestaurentManagement/src/main/java/com/restapi/models/Type.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+
```
Comment (vtronghieu) [inline RestaurentManagement/src/main/java/com/restapi/models/Type.java:18]: Pls remove this blank

**Comment by vtronghieu** [comment]
I saw you use manual getter-setter and constructors, this is not the best way. I know an annotation can help you do that. It means, that if you add that annotation, it will auto-generate a getter-setter, constructor,... for your class. You can research and try it.

File: RestaurentManagement/src/main/java/com/restapi/models/Menu.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    @Column(nullable = false,columnDefinition = "nvarchar(50)")
+    private String description;
+    @Column(nullable = false,columnDefinition = "nvarchar(150)")
+    private String image;
+    @Column(nullable = false)
+    private double price;
+    @OneToMany
+    private List<Type> type;
+
+    public Menu(UUID id, String name, String description, String image, double price, List<Type> type) {
```
Comment (vtronghieu) [inline RestaurentManagement/src/main/java/com/restapi/models/Menu.java:28]: You can research an annotation that can auto-generate this constructor

File: RestaurentManagement/src/main/java/com/restapi/models/Menu.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        this.id = id;
+        this.name = name;
+        this.description = description;
+        this.image = image;
+        this.price = price;
+        this.type = type;
+    }
+
+    public Menu() {
```
Comment (vtronghieu) [inline RestaurentManagement/src/main/java/com/restapi/models/Menu.java:37]: This constructor, too

File: RestaurentManagement/src/main/java/com/restapi/models/Type.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+@Data
+public class Type {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private UUID id;
+    @Column(nullable = false,columnDefinition = "nvarchar(50)")
+    private String type;
+
+    public Type(UUID id, String type) {
```
Comment (vtronghieu) [inline RestaurentManagement/src/main/java/com/restapi/models/Type.java:19]: Continue with this constructor

File: RestaurentManagement/src/main/java/com/restapi/models/Type.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        this.id = id;
+        this.type = type;
+    }
+
+    public Type() {
```
Comment (vtronghieu) [inline RestaurentManagement/src/main/java/com/restapi/models/Type.java:24]: And this constructor

File: RestaurentManagement/src/main/java/com/restapi/models/Menu.java (hunk 1 above)
Comment (PNH00) [reply RestaurentManagement/src/main/java/com/restapi/models/Menu.java]: done

File: RestaurentManagement/src/main/java/com/restapi/models/Type.java (hunk 2 above)
Comment (PNH00) [reply RestaurentManagement/src/main/java/com/restapi/models/Type.java:18]: done

File: RestaurentManagement/src/main/java/com/restapi/models/Menu.java (hunk 3 above)
Comment (PNH00) [reply RestaurentManagement/src/main/java/com/restapi/models/Menu.java:28]: maybe later

File: RestaurentManagement/src/main/java/com/restapi/models/Menu.java (hunk 4 above)
Comment (PNH00) [reply RestaurentManagement/src/main/java/com/restapi/models/Menu.java:37]: maybe later

File: RestaurentManagement/src/main/java/com/restapi/models/Type.java (hunk 5 above)
Comment (PNH00) [reply RestaurentManagement/src/main/java/com/restapi/models/Type.java:19]: maybe later

File: RestaurentManagement/src/main/java/com/restapi/models/Type.java (hunk 6 above)
Comment (PNH00) [reply RestaurentManagement/src/main/java/com/restapi/models/Type.java:24]: maybe later

**Pull request merged**